src/lock_file/update.rs

```diff
@@ -1025,12 +1028,18 @@ impl<'p> UpdateContextBuilder<'p> {
         // tool context
         let tool_context = ToolContext::builder()
             .with_gateway(gateway)
-            .with_client(client)
+            .with_client(client.clone())
             .build();
 
         let build_context =
             BuildContext::from_workspace(project)?.with_tool_context(Arc::new(tool_context));
 
+        let mapping_client = self.mapping_client.unwrap_or_else(|| {
+            MappingClient::builder(client)
+                .with_concurrency_limit(Arc::new(Semaphore::new(100)))
```

Comment:
After some testing I found that this could be really fast, but also extremely slow "sometimes" as in a single hyperfine run of 10 runs, it could differ between 0.7 second and 10 seconds. 

Lowering the value to 10 made it much more consistent between 1 and 2 seconds but it wouldn't go below 1 second anymore. Thus making the fastest run less fast. 

I think this should reuse the configuration setting `--concurrent-downloads` or we add `--concurrent-io-actions` and use that here too.

Reply:
I cannot reproduce any of these results on my windows machine. Ill try to reuse the `--concurrent-downloads` flag though!